src/component/card/card.jsx

```diff
@@ -1,7 +1,28 @@
-import { Link } from "react-router-dom";
+import { Link , useNavigate} from "react-router-dom";
+import { useCart } from "../../context/cartContext";
+import { useAuth } from "../../context/authContext";
```

Comment:
You can add an `index.js` file in your context folder and add all your context exports in that file, then you will be able to import all your contexts from a single file.